src/tests/functions-read-top-to-bottom.test.ts

```diff
@@ -0,0 +1,277 @@
+          const obj = {
+            firstMethod() {
+              return 'first';
+            },
+            secondMethod() {
+              return 'second';
+            }
+          };
+
+          function standalone() {
+            return obj.firstMethod();
+          }
+        `,
+      },
+    ],
+    invalid: [
+      {
+        code: `
+          function fetchData() {
+            return api.get('/data');
+          }
+
+          function processUserInput(input) {
+            return sanitize(input);
+          }
+
+          function transformData(data) {
+            return data.map((item) => item.value);
+          }
+
+          function handleClick() {
+            processUserInput(userInput);
+          }
+        `,
+        errors: [{ messageId: 'functionsReadTopToBottom' }],
+        output: `
+          function handleClick() {
+            processUserInput(userInput);
+          }
+
+function processUserInput(input) {
+            return sanitize(input);
+          }
+
+function fetchData() {
+            return api.get('/data');
+          }
+
+function transformData(data) {
+            return data.map((item) => item.value);
+          }
+        `,
+      },
+      {
+        code: `
+          const fetchData = () => {
+            return api.get('/data');
+          };
+
+          const processUserInput = (input) => {
+            return sanitize(input);
+          };
+
+          const transformData = (data) => {
+            return data.map((item) => item.value);
+          };
+
+          const handleClick = () => {
+            processUserInput(userInput);
+          };
+        `,
+        errors: [{ messageId: 'functionsReadTopToBottom' }],
+        output: `
+          const handleClick = () => {
+            processUserInput(userInput);
+          };
+
+const processUserInput = (input) => {
+            return sanitize(input);
+          };
+
+const fetchData = () => {
+            return api.get('/data');
+          };
+
+const transformData = (data) => {
+            return data.map((item) => item.value);
+          };
+        `,
+      },
+      {
+        code: `
+          function transformData(data) {
+            return data.map((item) => item.value);
+          }
+
+          function handleClick() {
+            processUserInput(userInput);
+          }
+
+          function fetchData() {
+            return api.get('/data');
+          }
+
+          function processUserInput(input) {
+            return sanitize(input);
+          }
+        `,
+        errors: [{ messageId: 'functionsReadTopToBottom' }],
+        output: `
+          function handleClick() {
+            processUserInput(userInput);
+          }
+
+function processUserInput(input) {
+            return sanitize(input);
+          }
+
+function fetchData() {
+            return api.get('/data');
+          }
+
+function transformData(data) {
+            return data.map((item) => item.value);
+          }
+        `,
```

Comment:
```suggestion
        output: `
          function transformData(data) {
            return data.map((item) => item.value);
          }
          
          function handleClick() {
            processUserInput(userInput);
          }

function processUserInput(input) {
            return sanitize(input);
          }

function fetchData() {
            return api.get('/data');
          }
        `,
```
We should only move nodes that need to be next to each other. We did not have to move these two nodes to the top of the file.